### PR TITLE
Adds a test to cover the toString method on CodeGenerationCollector

### PR DESCRIPTION
**Pull request by MelbourneDeveloper**
This PR adds a test for the CLI tool. It verifies that `CodeGenerationCollector.toString()` correctly outputs the valid error message. 

I found that this method was not covered by tests. I ran this script to output a report.

```bash
rm -rf ./coverage

dart pub get

dart pub global run coverage:test_with_coverage --package=.

lcov  ./coverage --output-file ./coverage/lcov.info --capture --directory

genhtml ./coverage/lcov.info --output-directory ./coverage/html 
```

After adding the test, the coverage showed that `toString()` was covered. This bumped coverage up from 76.9% to 77%.

Before:
image

After:

image

## Pre-launch Checklist

- [ x ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ x ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ x ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ x ] I added new tests to check the change I am making.
- [ x ] All existing and new tests are passing.
- [ x ] Any breaking changes are documented below.

## Breaking changes

None

### CODE REVIEW

File: tools/serverpod_cli/test/integration/analyzer/dart/endpoint_validation/endpoint_method_test.dart [hunk 1]
```diff
@@ -569,6 +570,20 @@ class ExampleEndpoint extends Endpoint {
       );
     });
 
+    test('then toString reports that the type is not supported.', () {
+      var expectedToString = '''Found 1 issue.
+
```
Comment (MelbourneDeveloper) [inline tools/serverpod_cli/test/integration/analyzer/dart/endpoint_validation/endpoint_method_test.dart:575]: This is in a sense, a snapshot test. It compares the output with a pre-calculated message. It doesn't need to be this verbose because changing the message format will break this test. However this test is comprehensive because it covers a lot of the potential corner cases of the output with not too much test code.

**Comment by MelbourneDeveloper** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by MelbourneDeveloper** [comment]
The integration test is failing on Windows. It turns out that the toString() output is slightly different between macOS and Windows. This is the Windows version:

```
Found 1 issue.

Error on line 6, column 31 of ${endpointFile.path}: The type "String Function()" is not a supported endpoint return type.
  ,
6 |   Future<TestFunctionBuilder> hello(Session session) async {
  |                               ^^^^^
  '
```

This is the macOS version:

```
Found 1 issue.

Error on line 6, column 31 of ${endpointFile.path}: The type "String Function()" is not a supported endpoint return type.
  ╷
6 │   Future<TestFunctionBuilder> hello(Session session) async {
  │                               ^^^^^
  ╵
```

**Comment by MelbourneDeveloper** [comment]
Just a note here. In some cases, I noticed that this file `endpointFile` is not closed by the end of the test on Windows. This didn't happen on macOS, but it happens occasionally on Windows. I believe that the test is not closing the file by the end and the delete fails because of this. This will probably cause brittleness in the test.

image

**Comment by SandPod** [comment]
Hey @MelbourneDeveloper!

Thanks for the contribution!
Test looks good and I'll do a last rebase and run on main before merging.

**Review by SandPod — APPROVED** [review]
🧪 🎉
